### ISSUE TITLE
Trusted host patterns? What?

### ISSUE THREAD

**Issue by ghost** [backdrop/backdrop-issues#4050]
I'm confused by "trusted host patterns". The only explanation is:

> specify a set of regular expressions that the domains on incoming requests must match.

(Source: https://api.backdropcms.org/documentation/trusted-host-settings)

What are incoming requests? It sounds like requests for my website. In which case, how on earth do I know what other sites/domains are linking to mine? Surely that's wrong and it means *my* website's domains (which at least I have a chance of knowing).

Also, what if I don't use this (like the whole time before this was available)? What are the risks of not using it, or of setting it up wrong?

I think some more documentation would be helpful...

**Comment by BWPanda** [comment]
Also added the 'needs - change notice' tag as there is text such as "link to change record?" on the page linked above, but no change notice I could find.

**Comment by herbdool** [comment]
Agreed it's confusing. It means basically that of you put your own domain name in there.

In settings.php it says you can disable the warning by setting it to false.

**Comment by herbdool** [comment]
Perhaps have an issue in the API documentation issue queue also.

**Comment by klonos** [comment]
I am taking a stab at a change record now. Will update here soon as I have a draft for it...

**Comment by findlabnet** [comment]
> In settings.php it says you can disable the warning by setting it to false.

Best approach so far.

**Comment by klonos** [comment]
Change record here: https://api.backdropcms.org/change-records/new-setting-for-trusted-hostname-configuration-optional-but-highly-recommended

>Also added the 'needs - change notice' tag as there is text such as "link to change record?" on the page linked above, but no change notice I could find.

I have also updated this to link to the change record 😉

**Comment by olafgrabienski** [comment]
**Copied from Gitter:**

@klonos:
...initially that page and the trusted host patterns one were a single thing, with the trusted host patterns section being at the end of that page. During feedback, it was pointed that because we were linking to that page from the admin UI (the warning in the site status page), it was bad UX to have the most important bit of text come after too much "literature" ...my quick fix this morning was to split that huge text to a parent + the trusted host patterns bit. Still needs work as you and @BWPanda have pointed...
...I plan to add another, 5th item in the list of the parent page, which points to the trusted host patterns page.

@olafgrabienski:
I'd suggest to have a single page again. Start with a short intro which summarizes the meaning of the huge text in one sentence, continue with the trusted host patterns part, and then put in the huge text at the end.
